src/GraphQL.MicrosoftDI/GraphQLBuilder.cs

```diff
@@ -0,0 +1,126 @@
+#nullable enable
+
+using System;
+using GraphQL.DI;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using MSServiceLifetime = Microsoft.Extensions.DependencyInjection.ServiceLifetime;
+using ServiceLifetime = GraphQL.DI.ServiceLifetime;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// An implementation of <see cref="IGraphQLBuilder"/> which uses the Microsoft dependency injection framework
+    /// to register services and configure options.
+    /// </summary>
+    internal class GraphQLBuilder : GraphQLBuilderBase
```

Comment:
internal sealed

Reply:
I really don’t know why you add sealed to all these internal classes. It makes no difference to users since they are internal classes. It doesn’t make any difference to us developers since we will just remove “sealed” if we derive from it. It also does not define scope, like private vs internal, so intellisense is unchanged. 

Sure, add sealed. It doesn’t matter. Personally I think it is cleaner without it.

Reply:
You won’t see internal sealed in any code I push.

Reply:
I also think that classes should as a general rule be non-sealed and have virtual methods. There are plenty of exceptions but it seems often that I come across an area where I want to inherit and cannot - like subscription support in the server project recently.

Reply:
Use `sealed` is just an old habit for me. For `sealed` classes compiler emits `call` instead of `callvirt`. Of course, it may vary from version to version, but the general meaning in optimization.

Reply:
> classes should as a general rule be non-sealed

I read the opposite judgments (from MS). The meaning is to make the class sealed and not allow it to change ahead of time without need. If/when needed, the class can be opened/unsealed. And if you open it too early, you will have to maintain backward compatibility because there may be descendants.

Reply:
This class was redefined to be public in a subsequent PR, so this comment does not apply.

I cannot at the moment remember why.  Perhaps it had to do with the server project, so it could add additional code in `Initialize()`.  But perhaps it was something else.

Reply:
I'll check.